### PR TITLE
178 firebase crash report

### PR DESCRIPTION
**Pull request by jl-skyline**
Integrate firebase crash report with demo application

### CODE REVIEW

File: MapboxAndroidDemo/build.gradle [hunk 1]
```diff
@@ -44,6 +44,8 @@ dependencies {
   compile 'com.getbase:floatingactionbutton:1.10.1'
   compile 'com.squareup.picasso:picasso:2.5.2'
 
+  compile 'com.google.android.gms:play-services:10.0.1'
```
Comment (cammace) [inline MapboxAndroidDemo/build.gradle]: This can just be `compile 'com.google.firebase:firebase-crash:10.0.0'` correct?

**Comment by cammace** [comment]
@tobrun @zugaldia, what is needed here to finish this?

**Comment by tobrun** [comment]
Thank you for picking up the requested changes @jlee6! 
I will push this branch to the repo to let our CI run and let CI import the services json.

**Comment by tobrun** [comment]
Been looking a bit more into the topic of only reporting crashes with release builds and there doesn't seem to be a straightforward solution. Multiple hacks shown in this [SO](http://stackoverflow.com/questions/37396826/how-to-disable-firebase-crash-reporting-when-the-app-is-running-on-debug) post. I'm going for a setup that introduces a different application id for debug builds (there is now 2 apps shown in the Mapbox-Android-Demo project in firebase). As a result we will need to keep google-services.json in the repo as we need it to run debug builds.

**Comment by tobrun** [comment]
Also noticing that `apply plugin: "com.google.gms.google-services" ` is required and reporting will not work without. To have this I need to enable multiDexSupport :disappointed:

**Comment by tobrun** [comment]
Couldn't get CI started from this external contribution.. Normally pulling the branch and pushing it should do this. To be able to merge I'm going to close this PR and follow up in #196

**Pull request closed without merging**